### PR TITLE
added takeTok, when and get to Token module

### PR DESCRIPTION
**Pull request by cryogenian**
I found this three function very useful in parsing token streams.

### CODE REVIEW

File: src/Text/Parsing/Parser/Token.purs [hunk 1]
```diff
@@ -1,14 +1,35 @@
...
 import Control.Monad.Error.Class
 
 import Text.Parsing.Parser
 import Text.Parsing.Parser.String
 import Text.Parsing.Parser.Combinators
 
+takeTok :: forall m a. (Monad m) => ParserT [a] m a
+takeTok = ParserT $ \s ->
+  return $ case s of
+    x:xs -> {consumed: true,
```
Comment (paf31) [inline src/Text/Parsing/Parser/Token.purs]: Could you please make this formatting consistent with the rest of the code?

File: src/Text/Parsing/Parser/Token.purs [hunk 2]
```diff
@@ -1,14 +1,35 @@
...
+             input: xs,
+             result: Right x}
+    _ -> {consumed: false,
+          input: s,
+          result: Left (ParseError {message: "there is nothing in 'takeTok'"})}
```
Comment (paf31) [inline src/Text/Parsing/Parser/Token.purs]: Suggested message: `Expected token, met EOF`.

File: src/Text/Parsing/Parser/Token.purs [hunk 3]
```diff
@@ -1,14 +1,35 @@
...
+
+when :: forall m a. (Monad m) => (a -> Boolean) -> ParserT [a] m a
+when f = try $ do
+  a <- takeTok
+  if f a then return a
```
Comment (paf31) [inline src/Text/Parsing/Parser/Token.purs]: Use `guard` here?

File: src/Text/Parsing/Parser/Token.purs [hunk 4]
```diff
@@ -1,14 +1,35 @@
...
+    else fail "token doesn't satisfy when test"
+
+get :: forall a m. (Monad m, Eq a) => a -> ParserT [a] m a
```
Comment (paf31) [inline src/Text/Parsing/Parser/Token.purs]: Can we call this `match` please?

File: src/Text/Parsing/Parser/Token.purs [hunk 5]
```diff
@@ -1,14 +1,35 @@
...
 
-import Control.Monad.State.Class
+import Control.Monad.State.Class hiding (get)
 import Control.Monad.Error.Class
 
 import Text.Parsing.Parser
 import Text.Parsing.Parser.String
 import Text.Parsing.Parser.Combinators
 
+takeTok :: forall m a. (Monad m) => ParserT [a] m a
```
Comment (paf31) [inline src/Text/Parsing/Parser/Token.purs]: Can we call this something like `head` or `token`? I generally don't like to abbreviate, especially when the word is so short to begin with :)

**Comment by paf31** [comment]
Code looks good, thanks, but I have some comments on style above. Could you please add code comments and possibly some tests too? Thanks.

**Comment by cryogenian** [comment]
Is it ok now?

**Comment by paf31** [comment]
Looks great, thank you!

**Pull request merged**